libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -774,6 +794,33 @@ public void onClick(DialogInterface dialog, int id) {
         dialog.show();
     }
 
+    public void showFeaturedImageConfirmationDialog(final int mediaId) {
+        GutenbergDialogFragment dialog = new GutenbergDialogFragment();
+        dialog.initialize(
+                TAG_REPLACE_FEATURED_DIALOG,
+                getString(R.string.replace_current_title),
+                getString(R.string.replace_current_description),
+                getString(R.string.replace_current_confirmation),
+                getString(R.string.replace_current_cancel),
+                mediaId
+        );
+
+        dialog.show(getChildFragmentManager(), TAG_REPLACE_FEATURED_DIALOG);
+    }
+
+    private void setFeaturedImage(int mediaId) {
+        mEditorFragmentListener.updateFeaturedImage(mediaId, false);
+        // Necessary to update image in Post Settings
+        setFeaturedImageId(mediaId);
```

Comment:
👋 Siobhan! Assuming the comment (`// Necessary to update image in the Post Settings`) is about the `setFeaturedImageId()` call, can you elaborate on the necessasity? I'm trying to follow the code and looks to me that [the member variable](https://github.com/wordpress-mobile/WordPress-Android/blob/74323e97c25ac24447dce5030f01f84f900eaf6b/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorFragmentAbstract.java#L143-L145) doesn't play a huge role and doesn't seem to be involved [when saving the data](https://github.com/wordpress-mobile/WordPress-Android/blob/332ca7f48d3142fae0267beecf0212166adc8fc6/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java#L999-L1020) 🤔. Maybe I'm missing a use?

Reply:
Hi @hypest to add to the conversation here, I can see where setting member variable would be useful if it's being persisted across configuration changes. 

For the "Necessary to update image in the Post Settings" comment, this will be removed once https://github.com/wordpress-mobile/WordPress-Android/pull/14521 is merged into this one. It is now under review.

Reply:
> would be useful if it's being persisted across configuration changes.

Hmm, the comment would still need some rephrasing then since it currently mentions "Post Settings" which makes me think of persisting in the local device prefs or the DB, which is different than preserving it on configuration changes (rotations for example). In any case, I feel that either the code doesn't correspond to the comment anymore, or vice versa 🙂

Reply:
It looks like the original bug referenced in the comment is no longer reproducible, but the functionality's still breaking if I comment out the call to `setFeaturedImageId`. I was able to reproduce the following with that line commented out:

* Toggle between **Set as Featured** and **Remove as Featured** on any image block, or between more than one image block.
* Notice how sometimes the button is unresponsive or the **replace** dialog pops up when there isn't actually a current featured image.

From logging, I see that `mFeaturedImageId` isn't always automatically updated when going through that flow. I'm not exactly sure if that's okay or expected. Is there another approach for updating that variable that should/could be taken here?

For now, I've updated the `Necessary to update image in Post Settings` to `Ensure "mFeaturedImageId" member variable is updated to specified "mediaId"`, in order to try to be clearer. Looking forward to hearing any extra thoughts!

Reply:
@hypest, we must have been drafting our comments at the same time and I missed yours until after I clicked published. :) You're right that the comment is now outdated, interested to hear what you think of the updated phrasing.

Reply:
> `Ensure "mFeaturedImageId" member variable is updated to specified "mediaId"`

Hmm, I guess the comment now is more accurate, but unfortunately ended up not adding any context. I mean, the `setFeaturedImageId(mediaId)` itself is nothing more than a classic (Java) setter so it is in fact already conveying the message "let's set the `FeaturedImageId` to `mediaId`. With a line comment, we are seeking to add more info that is not obvious just by looking at the code. So in this case, I'd expect a comment that would answer the question: "**Why** are we setting `mFeaturedImageId` to `mediaId`?"

I understand that perhaps we are not quite sure yet, and I guess that's what we need to do, to make sure we understand the code and the need. Let's continue reading and researching the code to see what's going and. That should also help answer your other question:
> "From logging, I see that mFeaturedImageId isn't always automatically updated when going through that flow. I'm not exactly sure if that's okay or expected. Is there another approach for updating that variable that should/could be taken here?".

Reply:
>  With a line comment, we are seeking to add more info that is not obvious just by looking at the code. So in this case, I'd expect a comment that would answer the question: "Why are we setting mFeaturedImageId to mediaId?"

@hypest, from my side, I'm not certain where the source of confusion is coming from. It feels correct to me that `setFeaturedImageId` would be called here, as I don't see anywhere else in the file where `mFeaturedImageId` is being updated. I wouldn't expect it to be automatically updated. It's clear I'm missing something here, but I'm not sure what?

Reply:
Ah, if there's no special meaning around the appearance of `setFeaturedImageId()` here (now that we've updated the original comment that was trying to elevate the importance) the we better just go ahead and remove the line comment as it doesn't add any context.

In other words, I'm not trying to say that the call should be somewhere else or not happening at all but, wondering if there's something special about it. If yes it makes sense to capture it in the line comment, otherwise it's actually better to remove the comment. Does that help?

Reply:
Got it, thank you for the extra explanation! This thread's definitely helped me to see how comments can cause confusion and this one is pretty unecessary given the function itself is fairly self-explanatory. I've gone ahead to remove it in https://github.com/wordpress-mobile/WordPress-Android/pull/14503/commits/36b6e2ee848e98baab166230f2e1de04366b92dd.